bugbug/repository.py

```diff
@@ -288,23 +288,29 @@ def hg_modified_files(hg, commit):
     )
 
 
-def get_touched_functions(path, deleted_lines, added_lines, content):
-    if content is None:
-        return set()
+def get_functions_from_metrics(metrics_space):
+    functions = []
+
+    if metrics_space["kind"] == "function" and metrics_space["name"] != "<anonymous>":
+        functions.append(
+            {
+                "end_line": metrics_space["end_line"],
+                "name": metrics_space["name"],
+                "start_line": metrics_space["start_line"],
+            }
+        )
+
+    for space in metrics_space["spaces"]:
+        functions += get_functions_from_metrics(space)
 
-    function_data = code_analysis_server.function(path, content)
-    if not function_data:
-        return set()
+    return functions
 
+
+def get_touched_functions(metrics_space, deleted_lines, added_lines):
```

Comment:
NOTE: I'm assuming the spaces are sorted, is that a safe assumption @calixteman?

Reply:
Yep, we make a DFS on the AST:
https://github.com/mozilla/rust-code-analysis/blob/master/src/metrics.rs#L470-L479